homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -367,7 +372,20 @@ async def async_get_battery(self):
                     server_info = await self.hass.async_add_executor_job(
                         self.leaf.get_latest_battery_status
                     )
-                    return server_info
+                    if (
+                        start_date is not None
```

Comment:
Could you simplify this to:
`if start_date and start_date != self._extract_start_date(server_info):`

Reply:
Yes I think you're right. I will make this change and test for a few hours.

Thanks!